internal/vault/configmap_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func TestOnShutDown(t *testing.T) {
+	ctx := context.Background()
+	cache, err := NewClientCache(100, nil, nil)
+	require.NoError(t, err)
+
+	clientFactory := cachingClientFactory{
+		cache:  cache,
+		logger: log.FromContext(ctx),
+	}
+
+	builder := fake.NewClientBuilder()
+	client := builder.Build()
+
+	tests := []struct {
+		cm       *corev1.ConfigMap
+		expected bool
+	}{
+		{
+			&corev1.ConfigMap{
+				Data: map[string]string{"anotherKey": "anotherValue"},
+			},
+			false,
+		},
+		{
+			&corev1.ConfigMap{
+				Data: map[string]string{ConfigMapKeyShutDownMode: ""},
+			},
+			false,
+		},
+		{
+			&corev1.ConfigMap{
+				Data: map[string]string{ConfigMapKeyShutDownMode: "invalidValue"},
+			},
+			false,
+		},
+		{
+			&corev1.ConfigMap{
+				Data: map[string]string{ConfigMapKeyShutDownMode: ShutDownModeRevoke.String()},
+			},
+			true,
+		},
+		{
+			&corev1.ConfigMap{
+				Data: map[string]string{ConfigMapKeyShutDownMode: ShutDownModeNoRevoke.String()},
+			},
+			true,
+		},
+	}
+
+	for _, tt := range tests {
+		onConfigMapChange := OnShutDown(&clientFactory)
+		actual := onConfigMapChange(ctx, client, tt.cm)
+		assert.Equal(t, tt.expected, actual)
+
+		if tt.expected {
```

Comment:
nit: I think we could use an assertion when `tt.expected=false` so that we are asserting correct behaviour on both paths.